packages/react-app/src/views/SubmissionReviewView.jsx

```diff
@@ -86,7 +86,12 @@ export default function SubmissionReviewView({ userProvider }) {
     // eslint-disable-next-line
   }, [address]);
 
-  const handleSendChallengeReview = reviewType => async (userAddress, challengeId, comment) => {
+  const handleSendChallengeReview = reviewType => async (badgeId, userAddress, challengeId, comment) => {
```

Comment:
🐛 Bug here

This changes the signature, but the calls are only updated when accepting the challenge, not when rejecting. That will cause the wrong params to be sent when rejecting. This breaks rejecting a challenge.

I get this error:
<img width=400 src="https://user-images.githubusercontent.com/17493200/144417132-3776378c-4105-455b-8478-61d11ef82a93.png" />

And this is the message we're signing:
<img width=400 src="https://user-images.githubusercontent.com/17493200/144417468-c1f5c9d3-af1b-4b58-89d7-6c341e9c6d0f.png"/>

Then, the challenge is not actually rejected. The backend throws an unhandled error and no event is saved in the database.

## Fix
I'll put the signature back as it was and revert the updated calls. I'll derive the badge id from the challenge id here instead of doing it on the calling point.

Reply:
Good catch!